search/includes/classes/class-search.php

```diff
@@ -885,13 +891,41 @@ public function get_statsd_request_mode_for_request( $url, $args ) {
 
 		// Bulk indexing
 		if ( '_bulk' === end( $path ) ) {
-			return 'index';
+			return $this->is_full_index( $path, $args ) ? 'full_index' : 'index';
 		}
 
 		// Unknown
 		return 'other';
 	}
 
+	/**
+	 * From the path will parse what type of indaxable is it. From the body of the request how many documents we are indexing.
+	 *
+	 * If the number of documents matches the number of corresponding objects in DB, will return true, false otherwise.
+	 */
+	private function is_full_index( array $path, array $args ) {
+		$body = $args['body'] ?? '';
+		$body_lines = explode( "\n", $body );
+
+		if ( $body_lines && is_array( $body_lines ) ) {
+			$doc_count = floor( count( $body_lines ) / self::LINES_PER_DOCUMENT_IN_BULK_INDEX );
+			$indexable_type_path = $path[ count( $path ) - 2 ];
+
+			$all_indexables = $this->indexables->get_all();
+			foreach ( $all_indexables as $indexable ) {
+				if ( strpos( $indexable_type_path, $indexable->slug ) !== false ) {
```

Comment:
I am not sure about this:
`strpos( $indexable_type_path, $indexable->slug ) !== false `

We basically check that the url has some slug in it. Any opinions? e.g. `vip-200508-post-1` contains `post`